### PR TITLE
<ServerName as TryFrom<_>> tweaks

### PR DESCRIPTION
**Pull request by djc**

### CODE REVIEW

File: src/server_name.rs [hunk 1]
```diff
@@ -74,6 +74,21 @@ impl<'a> fmt::Debug for ServerName<'a> {
     }
 }
 
+#[cfg(feature = "alloc")]
+impl TryFrom<String> for ServerName<'static> {
```
Comment (cpu) [inline src/server_name.rs:78]: I think we should update the `ServerName` doc block to reference this alongside the existing `ServerName::try_from(&str)`

**Review by cpu — APPROVED** [review]

**Review by cpu — APPROVED** [review]

File: src/server_name.rs (hunk 1 above)
Comment (djc) [reply src/server_name.rs:90]: Added. Please take a look; feel free to merge and publish if satisfied.

Comment (cpu) [reply src/server_name.rs:90]: Thanks, will do

**Comment by cpu** [comment]
* Pushed v/1.0.1 tag
* Published rustls-pki-types v1.0.1 at registry `crates-io`
* Created a [GitHub release](https://github.com/rustls/pki-types/releases/tag/v%2F1.0.1).

**Pull request merged**